test/CPUActivityAgentTest.cpp

```diff
@@ -538,3 +542,34 @@ TEST_F(CPUActivityAgentTest, no_mem_constconfig)
                                 "ConstConfigIOGroup configuration file does not contain memory bandwidth information");
 
 }
+
+TEST_F(CPUActivityAgentTest, invalid_fe)
```

Comment:
Caution: this test contains multiple tests inside, where each is a negative test that's testing a failed agent initialization, and they're all sharing the same agent object. I don't believe we make any guarantees about the state of an agent if ``init()`` fails, so these tests are depending on behavior that's not guaranteed (i.e. that a failed init leaves the agent object in a consistent state ready to attempt subsequent inits). I would recommend breaking this test out into multiple tests. If you want to keep it all in a single test, you could simply instantiate multiple agent objects.

Reply:
Broken out into 4 tests: invalid_fe_<core/uncore>_<high/low> to indicate which signal (core/uncore) fails & why (higher than range, lower than range).